### PR TITLE
with spring controllers

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@PropertySource("classpath:db.properties")
+@ComponentScan(basePackages = {
+        "cinema.service",
+        "cinema.dao"
+})
+public class AppConfig {
+    private Environment environment;
```
Comment (vladkuzya) [inline src/main/java/cinema/config/AppConfig.java:21]:
```suggestion
    private final Environment environment;
```

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+    @Autowired
+    public AuthenticationServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Autowired
+    public AuthenticationServiceImpl(ShoppingCartService shoppingCartService) {
+        this.shoppingCartService = shoppingCartService;
+    }
+
```
Comment (vladkuzya) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java:23]: Я думаю, что это можно соединить в один конструктор

File: src/main/java/cinema/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
 
+    @Autowired
+    public OrderServiceImpl(OrderDao orderDao) {
+        this.orderDao = orderDao;
+    }
+
+    public OrderServiceImpl(ShoppingCartService shoppingCartService) {
+        this.shoppingCartService = shoppingCartService;
+    }
+
```
Comment (vladkuzya) [inline src/main/java/cinema/service/impl/OrderServiceImpl.java:22]: the same

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -2,22 +2,28 @@
...
 
+    @Autowired
+    public ShoppingCartServiceImpl(ShoppingCartDao shoppingCartDao) {
+        this.shoppingCartDao = shoppingCartDao;
+    }
+
+    public ShoppingCartServiceImpl(TicketDao ticketDao) {
+        this.ticketDao = ticketDao;
+    }
+
```
Comment (vladkuzya) [inline src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:21]: and in all other cases

**Review by vladkuzya — COMMENTED** [review]
Nice work)

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -1,25 +1,31 @@
...
     private static final Logger logger = Logger.getLogger(AuthenticationServiceImpl.class);
-
-    @Inject
     private UserService userService;
-
-    @Inject
     private ShoppingCartService shoppingCartService;
 
+    @Autowired
+    public AuthenticationServiceImpl(UserService userService) {
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java]: will both userService and shoppingCartService be initialized, try to test it in main for example just for yourself?

File: src/main/java/cinema/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -1,24 +1,30 @@
...
 @Service
 public class OrderServiceImpl implements OrderService {
-    @Inject
     private OrderDao orderDao;
-
-    @Inject
     private ShoppingCartService shoppingCartService;
 
+    @Autowired
+    public OrderServiceImpl(OrderDao orderDao) {
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/service/impl/OrderServiceImpl.java]: same here

File: pom.xml [hunk 7]
```diff
@@ -43,6 +50,48 @@
...
+            <version>${spring.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
```
Comment (RomanDubovskyi) [inline pom.xml]: do we have any views now?

Comment (voloshynamaryna11) [reply pom.xml]: done

File: src/main/java/cinema/security/AuthenticationServiceImpl.java (hunk 5 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/security/AuthenticationServiceImpl.java]: done

File: src/main/java/cinema/service/impl/OrderServiceImpl.java (hunk 6 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/service/impl/OrderServiceImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**